name: Pull Request

# "This run was triggered by the "push" event, which is unsupported.
#  Please ensure you are using the "pull_request" event for this workflow."
on:
  pull_request:

concurrency:
  group: ${{ github.ref }}-${{ github.workflow }}
  cancel-in-progress: true

permissions:
  contents: read
  pull-requests: read

jobs:
  dependency-review:
    uses: percebus/github-actions-common/.github/workflows/dependency-review.yml@main

  pip-audit:
    # uses: percebus/github-actions-pipx/.github/workflows/run_pip-audit_detect.yml@main  # TODO

    runs-on: ubuntu-latest
    steps:
      - uses: percebus/github-actions-common/.github/actions/checkout@main
      - uses: percebus/github-actions-python/.github/actions/setup@main
      - uses: percebus/github-actions-pip/.github/actions/install-upgrade@main

      - name: pip-audit
        run: pipx run -- pip-audit . --locked

  #
  # package.json dependencies
  npm_audit:
    uses: percebus/github-actions-npm/.github/workflows/audit.yml@main
